lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -853,6 +853,7 @@ export class BrowserCacheManager extends CacheManager {
             }
         });
         this.clearMsalCookies();
+        this.setInteractionInProgress(false);
```

Comment:
Is this perhaps more an issue with `clearMsalCookies` not properly accounting for the fact that the interaction status key doesn't contain clientId anymore?

I'm wondering if always clearing both local/session storage and cookies here might cause new problems. For example there's an ongoing popup request but we called `cleanRequestByInteractionType("redirect")`. I would not expect to clear interaction in progress in this case.

Reply:
Should we instead update `clearMsalCookies` to specifically clear the interaction in progress cookie?

Reply:
I thought about that, this seemed a bit easier given we already have a method that understands how to clear the interaction in progress cache value.

Reply:
I'm more concerned about whether or not it's the right thing to clear session storage here (given that we're specifically trying to match interaction types). Were you able to determine how you can get into this state where the cookie still exists but session storage has been cleared?

Reply:
> I'm more concerned about whether or not it's the right thing to clear session storage here (given that we're specifically trying to match interaction types)

This should just clear the `msal.interaction.status` entry. That seems safe in this context (app is loaded and there is no hash in the url).

> Were you able to determine how you can get into this state where the cookie still exists but session storage has been cleared?

Linked ticket has the context, but I'm not sure how to reproduce it outside manually creating teh cookie.

Reply:
> This should just clear the msal.interaction.status entry. That seems safe in this context (app is loaded and there is no hash in the url)?

This function is called in a couple different contexts, not just on page load, so I don't think we can just assume it's ok to clear it.

Reply:
`cleanRequestByInteractionType` is only called one other place aside from the RedirectClient, which is here: 

https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/e3c66105f3bf617dc4fd06905e04b37e865c303c/lib/msal-browser/src/utils/PopupUtils.ts#L142

Reply:
Should this method check for interactiontype and only call `setInteractionProgress(false)` if the type is redirect?

Reply:
We can leave this as is for now but we should take a deeper look at our cache clearing logic in the future.

Reply:
I have the same concerns. That there should be more clarity for cookies vs session cookies.

Reply:
Does checking this in have side effects?

Reply:
Nothing comes to mind. I guess the question for me is that when we call `cealnRequestByInteractionType`, is there any reason we wouldn't want the interaction status to be cleared as well?

Reply:
Note that this value was getting cleared before, but then stopped getting cleared when the temp cache key for interaction status changed.

Reply:
> Nothing comes to mind. I guess the question for me is that when we call `cealnRequestByInteractionType`, is there any reason we wouldn't want the interaction status to be cleared as well?

A scenario I had in mind is if you call `loginPopup` and then while that is in progress you call `handleRedirectPromise`. We could argue that's incorrect usage and not likely something people would run into, though potentially a possible scenario with our wrapper libraries that call `handleRedirectPromise` in the background. This scenario would theoretically work today, but break with this change.

> Note that this value was getting cleared before, but then stopped getting cleared when the temp cache key for interaction status changed.

This value was getting cleared in **cookies** before, not necessarily also `sessionStorage` and I think that distinction is important for the scenario above.

Reply:
> A scenario I had in mind is if you call loginPopup and then while that is in progress you call handleRedirectPromise. We could argue that's incorrect usage and not likely something people would run into, though potentially a possible scenario with our wrapper libraries that call handleRedirectPromise in the background. This scenario would theoretically work today, but break with this change.

I thought about that, which is why I was curious if I should add a check to only clear interaction status here if the interaction type is redirect.

> This value was getting cleared in cookies before, not necessarily also sessionStorage and I think that distinction is important for the scenario above.

Was this nuance intentional?

Reply:
> I thought about that, which is why I was curious if I should add a check to only clear interaction status here if the interaction type is redirect.

That wouldn't solve the problem since `handleRedirectPromise` would still clear an ongoing popup interaction.

> Was this nuance intentional?

I don't remember specifically why this was but I suspect it's because request cookies are only important for redirect scenarios so it doesn't matter what the interaction type is.

Reply:
> That wouldn't solve the problem since handleRedirectPromise would still clear an ongoing popup interaction.

I only see references to `this.browserStorage.cleanRequestByInteractionType(InteractionType.Redirect);` in `handleRedirectPromise`?

> I don't remember specifically why this was but I suspect it's because request cookies are only important for redirect scenarios so it doesn't matter what the interaction type is.

Ah, makes sense. But as you note, invoking `handleRedirectPromise` during a popup operation is not correct, so this clearing session storage shouldn't be an issue, yeah?